app/controllers/account_subscriptions_controller.rb

```diff
@@ -74,4 +76,16 @@ def reauthenticate_user
       ),
     )["auth_uri"]
   end
+
+  def has_subscription?(subscriber)
+    subscriptions = GdsApi.email_alert_api.get_subscriptions(id: subscriber.fetch("id")).to_h.fetch("subscriptions", [])
+
+    !subscriptions.find { |s| s["subscriber_list"]["id"] == @subscriber_list["id"] }.nil?
+  end
+
+  def redirect_with_already_subscribed
+    account_flash_add CreateAccountSubscriptionService::ALREADY_SUBSCRIBED_FLASH
+
+    redirect_to @subscriber_list["url"]
```

Comment:
I think this should be `redirect_with_analytics`, to handle the case:

1. A user clicks the button on a page while logged out
2. They get redirected to DI to sign in
3. They get redirected to this endpoint from DI after signing in, with analytics parameters
4. We then redirect them back to the content page, and should preserve those analytics parameters because the javascript has not yet run